src/library_nodefs.js

```diff
@@ -269,17 +269,18 @@ addToLibrary({
         }
       },
       read(stream, buffer, offset, length, position) {
+        //err(`nodefs read: offset=${offset} length=${length} position=${position}`);
         // Node.js < 6 compatibility: node errors on 0 length reads
         if (length === 0) return 0;
         try {
-          return fs.readSync(stream.nfd, Buffer.from(buffer.buffer), offset, length, position);
+          return fs.readSync(stream.nfd, new Int8Array(buffer.buffer, offset, length), { position: position });
```

Comment:
This I'm not clear on.. some kind of internal node limit is reached with the existing code.

I'll add a comment.

Reply:
interesting... is this limitation shared by the browser? IIRC the way array buffers are allocated is handled by the embedder rather than V8 proper so it might be different.

Reply:
I added a comment to the PR description.. its basically the same v8 issue we work around elsewhere related to size of an array view.